src/hero_bringup/toyota_software_utils.py

```diff
@@ -0,0 +1,108 @@
+import os
+import rospkg.common
+import rospkg.rospack
+from xml.etree.ElementTree import ElementTree
+import yaml
+
+global RP
+RP = None
+
+
+def read_dict_yaml(file_path, log):
+    # type: (str, callable) -> dict
+    """
+    Read a yaml file, which should be a dict at toplevel
+
+    :param file_path: yaml file to read
+    :type file_path: str
+    :param log: Log function to use, must accept a string
+    :type log: callable
+    :return: yaml object of dictionaries and lists
+    :rtype: dict
+    :raises yaml.parser.ParserError, yaml.scanner.ScannerError: In case of invalid yaml syntax
+    """
+
+    with open(file_path) as f:
+        try:
+            yaml_obj = yaml.load(f, yaml.CSafeLoader)
+        except AttributeError:
+            yaml_obj = yaml.load(f, yaml.SafeLoader)
+        except (yaml.parser.ParserError, yaml.scanner.ScannerError) as e:
+            log("Invalid yaml syntax: {0}".format(e))
+            raise
+
+    return yaml_obj
+
+
+def get_rospack():
```

Comment:
I get why you do this (at least, I guess it gets slow if you don't). Maybe add that as a comment